### PR TITLE
Add warning about trailing forward slash

### PR DESCRIPTION
**Pull request by meffij**

### CODE REVIEW

**Comment by ResidentMario** [comment]
Two comments:

1. This probably belongs in the Issues section, alongside the rest of the warnings about rough edges.
2. As I understand it the problem isn't pushing anything ending in `/` to T4. The problem is pushing *files* ending in `/` to T4. If I push a *directory* to a key ending in `/`, then everything works fine. And indeed we do this, and recommending doing this, in `QuickStart.ipynb`.

@akarve what are your thoughts?

File: UserDocs.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ Writes the in-memory object `src` to the path `dest` in S3.
 Transparently serializes some object types.
 See [Serialization](#serialization) for details.
 
+`dest` should not end in `/`. Objects that end in `/` are treated specially by some s3 tools in a way that is potentially dangerous, so it's best to avoid them.
```
Comment (akarve) [inline UserDocs.md]: not the right place to document this - because `he.put()` automatically forbids it

File: UserDocs.md [hunk 2]
```diff
@@ -55,6 +57,8 @@ Writes the in-memory object `src` to the path `dest` in S3.
...
```
Comment (akarve) [inline UserDocs.md]: Also not the right place to document this as the API only allows `dest/` if you have `src/` and `src/` is a dir? if that's not the case we should make it the case :)

File: UserDocs.md [hunk 3]
```diff
@@ -55,6 +57,8 @@ Writes the in-memory object `src` to the path `dest` in S3.
...
+
```
Comment (akarve) [inline UserDocs.md]: My general feedback. This bullet belongs in the "Known Issues" section, with a link to the corresponding AWS docs. Does not belong in API docs b/c API should forbid this case whenever possible?

**Comment by akarve** [comment]
@ResidentMario agreed - just check that `put_file(src, dest/)` throws an exception

File: UserDocs.md [hunk 4]
```diff
@@ -353,3 +353,6 @@ T4 automatically populates the following metadata:
     s3:GetObject
     s3:GetObjectVersion
     ```
+* The keys of objects in s3 should not end in `/`. Objects that end in `/` are treated specially by some s3 tools in a way that is potentially dangerous, so it's best to avoid them. The helium API will help you avoid this rough edge by rejecting keys that end in `/`.
```
Comment (ResidentMario) [inline UserDocs.md]: We capitalize `s3` as `S3` throughout the rest of the doc, we should do that here too.

File: UserDocs.md [hunk 5]
```diff
@@ -353,3 +353,6 @@ T4 automatically populates the following metadata:
...
+
+Refer to [Amazon's documentation](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/using-folders.html) on folder objects.
```
Comment (ResidentMario) [inline UserDocs.md]: I would make this the last sentence in the paragraph above. Right now it's hanging out on its own line below the rest of the text in the rendered text on GitHub.

File: UserDocs.md [hunk 6]
```diff
@@ -353,3 +353,8 @@ T4 automatically populates the following metadata:
     s3:GetObject
     s3:GetObjectVersion
     ```
+* The keys of objects in S3 should not end in `/`. Objects that end in `/`
```
Comment (akarve) [inline UserDocs.md]: "Objects whose keys end in `/`

File: UserDocs.md [hunk 7]
```diff
@@ -353,3 +353,8 @@ T4 automatically populates the following metadata:
...
+are treated specially by some S3 tools in a way that
+is potentially dangerous, so it's best to avoid them.
+The helium API will help you avoid this rough edge by rejecting keys that end in `/`.
```
Comment (akarve) [inline UserDocs.md]: "by rejecting object keys that end in `/`"
(i.e. folder keys that end in `/` are legit and we should probably spell that out)

**Pull request merged**